File: coco/src/main/java/coco/ep/m6/Exp.java
package coco.ep.m6;

public interface Exp<FT> extends coco.ep.m5.Exp<FT>, Factory<FT> {

    default Boolean isLit(Double d) { return false; }
    default Boolean isAdd(coco.ep.Exp<FT> left, coco.ep.Exp<FT> right) { return false; }
    default Boolean isSub(coco.ep.Exp<FT> left, coco.ep.Exp<FT> right) { return false; }
    default Boolean isMult(coco.ep.Exp<FT> left, coco.ep.Exp<FT> right) { return false; }
    default Boolean isNeg(coco.ep.Exp<FT> inner) { return false; }
    default Boolean isDivd(coco.ep.Exp<FT> left, coco.ep.Exp<FT> right) { return false; }

    Boolean equals(coco.ep.Exp<FT> other);
    Boolean eql(coco.ep.Exp<FT> other);
}

File: coco_csharp/CoCo/EP/M6/Exp.cs
namespace EP.M6 {
    public interface Exp<FT> : Factory<FT>, EP.M5.Exp<FT>  {
        bool isLit(double d) { return false; }
        bool isAdd(EP.Exp<FT> left, EP.Exp<FT> right) { return false; }
        bool isSub(EP.Exp<FT> left, EP.Exp<FT> right) { return false; }
        bool isMult(EP.Exp<FT> left, EP.Exp<FT> right) { return false; }
        bool isNeg(EP.Exp<FT> inner) { return false; }
        bool isDivd(EP.Exp<FT> left, EP.Exp<FT> right) { return false; }

        bool equals(EP.Exp<FT> other);
        bool eql(EP.Exp<FT> other);
    }
}